### PR TITLE
Travis and Appveyor testing for OSX/Ubuntu/Windows

### PR DESCRIPTION
**Pull request by hmaarrfk**
Would you consider having Travis and Appveyor test this package automatically? I mostly pull things from the instructions in the README.md, so I test those too ;).

I've tested that this work:
https://github.com/hmaarrfk/miniver/pull/1

But you would need to sign up and enable Travis and Appveyor for your repo.

Your script actually works on windows, so I updated the Readme message.

We can add travis and appveyor badges too if you are so inclined.

Awesome stuff BTW.

### CODE REVIEW

**Comment by jbweston** [comment]
Awesome, thanks!

> Your script actually works on windows, so I updated the Readme message.

Nice!

I'll take a look at this now

File: ci/create_package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+with open('../miniver/README.md') as f:
+    for line in f:
+        if line.startswith("# Your package's __init__.py"):
+            break
+    with open('my_package/__init__.py', 'w+') as out:
+        out.write(line)
+        for line in f:
+            if line.startswith('```'):
+                break
+            out.write(line)
```
Comment (jbweston) [inline ci/create_package.py]: Maybe we could make this block into a function; something like:

```python
def write_snippet_from_readme(start_marker, outfile):
    ...

...

write_snippet_from_readme("# Your project's setup.py", 'setup.py')
write_snippet_from_readme("# Your project's .gitattributes", '.gitattributes')
write_snippet_from_readme("# Your package's __init__.py", 'my_package/__init__.py')
```

What do you think?

**Comment by hmaarrfk** [comment]
Also. I don’t want to hold you to supporting windows if you don’t want to. I can revert the read me and you can add something to your liking.

Including a windows appveyor test would mostly allow others to contribute in that department if something breaks windows. I also gives me confidence that miniver is a good substitute for versioneer.

**Comment by jbweston** [comment]
> Also. I don’t want to hold you to supporting windows if you don’t want to. I can revert the read me and you can add something to your liking.

It seems reasonable as long as it doesn't require a bunch of special cases etc.

File: ci/create_package.py (hunk 1 above)
Comment (hmaarrfk) [reply ci/create_package.py]: Right, we could. We would need to find a way to add the line
```
from setuptools import setup
```
to setup.py. I think it is dooable through a second parameter.

Comment (jbweston) [reply ci/create_package.py]: I would say just add that explicitly:

```
with open('setup.py', 'w+') as f:
    f.writelines('from setuptools import setup')

write_snippet_from_readme("# Your project's setup.py", 'setup.py')
...
```

This special case doesn't seem so special as to deserve an extra parameter IMO

**Comment by hmaarrfk** [comment]
I also noticed that I don't test the sdist or the bdist. For that, I think we need to update the setup.py snippet to include `packages='my_package'` or `packages=find_packages()`.

**Comment by jbweston** [comment]
> For that, I think we need to update the setup.py snippet to include packages='my_package' or packages=find_packages()

We can do that now if you are motivated, or we can do that in a followup PR, since this one is already very useful on its own.

**Comment by hmaarrfk** [comment]
Followup is probably best.

I just wanted to put it out there.

I don't really have time to work on this for a while anyway.

**Comment by jbweston** [comment]
:+1:

**Pull request merged**